src/main/webapp/script.js

```diff
@@ -11,3 +11,48 @@
 // WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 // See the License for the specific language governing permissions and
 // limitations under the License.
+
+/*
+ * Displays navbar authentication buttons according to login status.
+ */
+window.onload = function fetchLogin() {
+  fetch('/login').then(response => response.json()).then(user => {
+    // If user is logged in, show logout button in navbar.
+    if (user.loggedIn == true) {
+      // Delete signup button.
+      const signupButtonNavbar = document.getElementById('signup');
+      signupButtonNavbar.innerHTML = '';
+
+      // Add logout button to navbar.
+      addAuthenticationButton(user.authenticationUrl, 'btn-outline-success', 'Log Out', 'login');
+    // If user is logged out, show signup and login buttons in navbar.
+    } else {
+      // Add signup button to navbar.
+      addAuthenticationButton(user.authenticationUrl, 'btn-success', 'Sign Up', 'signup');
+
+      // Add login button to navbar.
+      addAuthenticationButton(user.authenticationUrl, 'btn-outline-success', 'Log In', 'login');
+    }
+  })
+}
+
+function addAuthenticationButton(authenticationUrl, buttonStyle, buttonText, navbarItem) {
```

Comment:
Awesome change, looks so much better :) Mind adding a small method-level comment?

Reply:
I accidentally made the change in a branch that branches off from this one, but the change is made. I'll just keep it like this to avoid merge conflicts later, but it is added.

Reply:
This is the comment, so you don't look for the branch:
```
/**
 * Creates a signup, login, or logout button and appends it to navbar.
 * @param {string} authenticationUrl 
 * @param {string} buttonStyle 
 * @param {string} buttonText 
 * @param {string} navbarItem 
 */
```